pages/admin/reports/components/TotalEarnings.tsx

```diff
@@ -0,0 +1,260 @@
+import { Box, Text, useBreakpointValue } from '@chakra-ui/react';
+import {
+    BarChart,
+    ResponsiveContainer,
+    Bar,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+} from 'recharts';
+import { useEffect, useState } from 'react';
+import { getAppointments } from '@/services/AppointmentService';
+import { Appointment } from '@/interfaces/appointment/Appointment';
+import { splitDateTimeFromISO } from '@/utils/dateUtils';
+import {
+    TotalEarningsProps,
+    TreatmentStats,
+} from '@/interfaces/statistics/TotalEarnings';
+import parseTreatmentPrice from '@/utils/priceUtils';
+
+export default function TotalEarnings({
+    statistics,
+    dateRange,
+    selectedTreatmentId,
+    treatments,
+}: TotalEarningsProps) {
+    const [appointments, setAppointments] = useState<Appointment[]>([]);
+    const [filteredStats, setFilteredStats] = useState<TreatmentStats[]>([]);
+    const isMobile = useBreakpointValue({
+        base: true,
+        sm: true,
+        md: true,
+        lg: false,
+    });
+
+    const formatTreatmentName = (name: string) => {
+        if (!name) return '';
+        if (!isMobile) {
+            return name;
+        }
+        // If the name has spaces, use initials
+        const words = name.split(' ');
+        if (words.length > 1) {
+            return words.map((word) => word.charAt(0)).join('');
+        }
+        return name.length > 6 ? name.substring(0, 5) + '.' : name;
+    };
+
+    // Load appointments
+    useEffect(() => {
+        if (!statistics) {
+            setAppointments([]);
+            setFilteredStats([]);
+            return;
+        }
+        const fetchAppointments = async () => {
+            try {
+                const data = await getAppointments();
+                setAppointments(data);
+            } catch (error) {
+                console.error('Error al cargar citas:', error);
+            }
+        };
+        fetchAppointments();
+    }, [statistics]);
+
+    useEffect(() => {
+        if (!dateRange?.startDate || !dateRange.endDate) {
+            setFilteredStats([]);
+        }
+    }, [dateRange?.startDate, dateRange?.endDate]);
+
+    useEffect(() => {
+        if (!appointments.length || !treatments.length) {
```

Comment:
Idk if something liek this could save you further issues with date verifications:
```suggestion
        if (!appointments.length || !treatments.length || !dateRange?.startDate || !dateRange?.endDate) {
    setFilteredStats([]);
    return;
}
```